source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2557,46 +2555,29 @@ void COpenGLDriver::setBasicRenderStates(const SMaterial& material, const SMater
 	}
 
 	// Anti aliasing
-	if (resetAllRenderStates || lastmaterial.AntiAliasing != material.AntiAliasing)
-	{
-		if (FeatureAvailable[IRR_ARB_multisample])
+	if (resetAllRenderStates
+			|| lastmaterial.AntiAliasing != material.AntiAliasing
+			&& FeatureAvailable[IRR_ARB_multisample]) {
```

Comment:
on second look this is actually broken
needs to be `(A || B) && C`

Reply:
You're right; thanks.